### PR TITLE
Fix for bug #300 - incorrect category node output by resource containing embedded conditional links

### PR DESCRIPTION
**Pull request by dlgroves**

### CODE REVIEW

File: interaction-core/src/main/java/com/temenos/interaction/core/resource/EntityResource.java [hunk 1]
```diff
@@ -129,4 +128,22 @@ public String getEntityTag() {
 	public void setEntityTag(String entityTag) {
 		this.entityTag = entityTag;
 	}
+
+	/**
+	 * Create a shallow copy of this EntityResource using the same entity, links and etag
+	 * as this instance.
```
Comment (mtemenos) [inline interaction-core/src/main/java/com/temenos/interaction/core/resource/EntityResource.java]: What was the benefit of using clone over say a copy constructor or static factory method?

Comment (dlgroves) [inline interaction-core/src/main/java/com/temenos/interaction/core/resource/EntityResource.java]: We preferred clone() as we felt that this was more testable than a copy constructor in terms of verifying interactions and also felt that using clone() encapsulated this task better than another constructor.

Comment (mtemenos) [inline interaction-core/src/main/java/com/temenos/interaction/core/resource/EntityResource.java]: and factory method?

Comment (dlgroves) [inline interaction-core/src/main/java/com/temenos/interaction/core/resource/EntityResource.java]: A factory method would have been more difficult to test in this task than a clone() method as static methods cannot easily be mocked.

**Pull request merged**